multisig_wallet/multisig_wallet.py

```diff
@@ -373,26 +388,28 @@ def getConfirmations(self, _offset: int, _count: int, _transactionId: int) -> li
     def getTransactionCount(self, _pending: bool = True, _executed: bool = True) -> int:
         tx_count = 0
         for tx_id in range(self._transaction_count.get()):
-            if (_pending and not self._transactions[tx_id][0]) or (_executed and self._transactions[tx_id][0]):
+            if (self._transactions[tx_id] is not None) and (
+                    (_pending and not self._transactions[tx_id][0])
+                    or (_executed and self._transactions[tx_id][0])):
                 tx_count += 1
-
         return tx_count
 
     @external(readonly=True)
     def getTransactionList(self, _offset: int, _count: int, _pending: bool = True, _executed: bool = True) -> list:
-        self._only_positive_number(_offset, _count)
+        self._check_0_and_above(_offset, _count)
 
         if _count > self._MAX_DATA_REQUEST_AMOUNT:
-            revert("requests that exceed the allowed amount")
+            revert("Exceed max of requests")
 
         transaction_list = []
-        total_transaction_count = self._transaction_count.get()
+        transaction_count = self._transaction_count.get()
 
-        # prevent searching not existed transaction
-        _count = _offset + _count if total_transaction_count >= _offset + _count else total_transaction_count
-
-        for tx_id in range(_offset, _count):
-            if (_pending and not self._transactions[tx_id][0]) or (_executed and self._transactions[tx_id][0]):
+        for tx_id in range(_offset, transaction_count):
+            if len(transaction_list) == _count:
```

Comment:
I guess This logic is needed to be fixed. In this logic, '_count' is used to check the count of transactions, but this is not my intention.